extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableParquetLocationKey.java

```diff
@@ -48,4 +61,36 @@ public String getImplementationName() {
     public ParquetInstructions readInstructions() {
         return readInstructions;
     }
+
+    /**
+     * Precedence-wise this implementation compares {@code order}, then {@code dataSequenceNumber}, then
+     * {@code fileSequenceNumber}, then {@code pos}, then applies a {@link PartitionsComparator} to {@code partitions}
+     * and finally {@code uri}.
+     *
+     * @inheritDoc
+     */
+    @Override
+    public int compareTo(@NotNull final TableLocationKey other) {
+        if (other instanceof IcebergTableParquetLocationKey) {
+            final IcebergTableParquetLocationKey otherTyped = (IcebergTableParquetLocationKey) other;
+            int comparisonResult;
+            if ((comparisonResult = Integer.compare(order, otherTyped.order)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = Long.compare(dataSequenceNumber, otherTyped.dataSequenceNumber)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = Long.compare(fileSequenceNumber, otherTyped.fileSequenceNumber)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = Long.compare(pos, otherTyped.pos)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = PartitionsComparator.INSTANCE.compare(partitions, otherTyped.partitions)) != 0) {
+                return comparisonResult;
+            }
+            return uri.compareTo(otherTyped.uri);
```

Comment:
Usually when a comparator is changed like this, we need to ensure that equals (and to a lesser extend hashcode) are updated to be consistent.

I see that the super class just uses uri and partitions for equals and hashcode; I think _normally_ this would be good enough, but I wonder if it's ever possible to use the same uri (and in the worst case, the comparisons above are all equal or all null); in which case, maybe we need to actually think use ManifestFile as part of equality / comparison logic?

I think we may need to consider a parquet location key as ManifestFile + DataFile?

Reply:
So ManifestFile provides an additional 
```ManifestFile::sequenceNumber() : Returns the sequence number of the commit that added the manifest file.```

I can add that to the mix, after URI.